src/test/tst_attributerange.cpp

```diff
@@ -30,10 +31,632 @@ class TestAttributeRange: public QObject
+
+    QCOMPARE(attrRge->min().toBool(), min_int);
+
+//    QCOMPARE(attrRge->max().toBool(), max_int);
+    QCOMPARE(attrRge->max().toBool(), false);
+
+    delete prg;
+}
+
+void TestAttributeRange::tst_int_range()
+{
+    QStringList attrNames = {"test0", "test1", "test2"};
+    // Case 1: regular range
+    // Case 2: negative numbers
+    // Case 3: large range
+    QStringList attrRangeStrs = {"int[0,1]", "int[-10,-5]", "int[-100,100]"};
+
+    AttributeRange* attrRge = AttributeRange::parse(0, attrNames[0], attrRangeStrs[0]);
+    AttributeRange* attrRge2 = AttributeRange::parse(1, attrNames[1], attrRangeStrs[1]);
+    AttributeRange* attrRge3 = AttributeRange::parse(2, attrNames[2], attrRangeStrs[2]);
+
+    // Tests value returned by 'AttributeRange::validate()'
+    Value v;
+
+    v = attrRge->validate("0");   // min
+    _tst_value(v, Value::INT);
+    v = attrRge->validate("1");   // max
+    _tst_value(v, Value::INT);
+    v = attrRge->validate("invalid"); // invalid
+    _tst_value(v, Value::INVALID);
+
+    v = attrRge2->validate("-10");   // min
+    _tst_value(v, Value::INT);
+    v = attrRge2->validate("-5");   // max
+    _tst_value(v, Value::INT);
+    v = attrRge2->validate("-7");   // other value
+    _tst_value(v, Value::INT);
+    v = attrRge2->validate("invalid"); // invalid
+    _tst_value(v, Value::INVALID);
+
+    v = attrRge3->validate("-100");   // min
+    _tst_value(v, Value::INT);
+    v = attrRge3->validate("100");   // max
+    _tst_value(v, Value::INT);
+    v = attrRge3->validate("0");   // other value
+    _tst_value(v, Value::INT);
+    v = attrRge3->validate("invalid"); // invalid
+    _tst_value(v, Value::INVALID);
+
+    // Tests if functions work as expected
+    QVERIFY(attrRge->isValid());
+    QCOMPARE(attrRge->id(), 0);
+    QCOMPARE(attrRge->attrName(), attrNames.at(0));
+    QCOMPARE(attrRge->attrRangeStr(), attrRangeStrs.at(0));
+
+    QVERIFY(attrRge2->isValid());
+    QCOMPARE(attrRge2->id(), 1);
+    QCOMPARE(attrRge2->attrName(), attrNames.at(1));
+    QCOMPARE(attrRge2->attrRangeStr(), attrRangeStrs.at(1));
+
+    QVERIFY(attrRge3->isValid());
+    QCOMPARE(attrRge3->id(), 2);
+    QCOMPARE(attrRge3->attrName(), attrNames.at(2));
+    QCOMPARE(attrRge3->attrRangeStr(), attrRangeStrs.at(2));
+
+    AttributeRange::Type type = AttributeRange::Int_Range;
+    QCOMPARE(attrRge->type(), type);
+    QCOMPARE(attrRge2->type(), type);
+    QCOMPARE(attrRge3->type(), type);
+
+    // Tests min(), max() and rand() functions
+    int min, max;
+    PRG* prg = new PRG(123);
+
+    min = 0;
+    max = 1;
+    QCOMPARE(attrRge->min().toInt(), min);
+    QCOMPARE(attrRge->max().toInt(), max);
+    v = attrRge->rand(prg);
+    _tst_value(v, Value::INT);
+    QVERIFY(v.toInt() >= min);
+    QVERIFY(v.toInt() <= max);
+
+    min = -10;
+    max = -5;
+    QCOMPARE(attrRge2->min().toInt(), min);
+    QCOMPARE(attrRge2->max().toInt(), max);
+    v = attrRge2->rand(prg);
+    _tst_value(v, Value::INT);
```

Comment:
As we won't have the `_tst_value` anymore, just checking the type should be fine here, i.e.,
``` c++
QCOMPARE(v.type(), Value::INT);
```
as we have all tests for a `Value`, we can trust that if `type()==Value::INT` then `v.isInt()` is the only one to return `true`.